recipes-bsp/imx-mkimage/imx-boot_1.0.bb

```diff
@@ -233,29 +315,56 @@ deploy_mx93() {
     fi
 }
 
+deploy_mx93_extra() {
+    if [ -e ${DEPLOY_DIR_IMAGE}/u-boot-spl.bin-${MACHINE}-${UBOOT_CONFIG} ] ; then
+        install -m 0644 ${DEPLOY_DIR_IMAGE}/u-boot-spl.bin-${MACHINE}-${UBOOT_CONFIG_EXTRA} \
+                                                             ${DEPLOYDIR}/${BOOT_TOOLS}
+    fi
+}
+
 do_deploy() {
     deploy_${SOC_FAMILY}
-    # copy the sc fw, dcd and uboot to deploy path
-    install -m 0644 ${DEPLOY_DIR_IMAGE}/${UBOOT_NAME}        ${DEPLOYDIR}/${BOOT_TOOLS}
 
     # copy tee.bin to deploy path
     if ${DEPLOY_OPTEE}; then
        install -m 0644 ${DEPLOY_DIR_IMAGE}/tee.bin ${DEPLOYDIR}/${BOOT_TOOLS}
     fi
-
     # copy makefile (soc.mak) for reference
     install -m 0644 ${BOOT_STAGING}/soc.mak                  ${DEPLOYDIR}/${BOOT_TOOLS}
-    # copy the generated boot image to deploy path
-    for target in ${IMXBOOT_TARGETS}; do
-        # Use first "target" as IMAGE_IMXBOOT_TARGET
-        if [ "$IMAGE_IMXBOOT_TARGET" = "" ]; then
-            IMAGE_IMXBOOT_TARGET="$target"
-            echo "Set boot target as $IMAGE_IMXBOOT_TARGET"
+
+    for type in ${UBOOT_CONFIG}; do
+        UBOOT_CONFIG_EXTRA="$type"
+        UBOOT_NAME_EXTRA="u-boot-${MACHINE}.bin-${UBOOT_CONFIG_EXTRA}"
+        BOOT_CONFIG_MACHINE_EXTRA="${BOOT_NAME}-${MACHINE}-${UBOOT_CONFIG_EXTRA}.bin"
+
+        deploy_${SOC_FAMILY}_extra
+        # copy the tool mkimage to deploy path and sc fw, dcd and uboot
+        install -m 0644 ${DEPLOY_DIR_IMAGE}/${UBOOT_NAME_EXTRA}  ${DEPLOYDIR}/${BOOT_TOOLS}
+
+        # copy the generated boot image to deploy path
+        for target in ${IMXBOOT_TARGETS}; do
+            # Use first "target" as IMAGE_IMXBOOT_TARGET
+            if [ "$IMAGE_IMXBOOT_TARGET" = "" ]; then
+                IMAGE_IMXBOOT_TARGET="$target"
+                echo "Set boot target as $IMAGE_IMXBOOT_TARGET"
+            fi
+            install -m 0644 ${S}/${BOOT_CONFIG_MACHINE_EXTRA}-${target} ${DEPLOYDIR}
+        done
+
+        # The first UBOOT_CONFIG listed will be the ${BOOT_NAME} binary
+        if [ ! -f "${DEPLOYDIR}/${UUU_BOOTLOADER}" ]; then
+            cd ${DEPLOYDIR}
+            ln -sf ${BOOT_CONFIG_MACHINE_EXTRA}-${IMAGE_IMXBOOT_TARGET} ${BOOT_NAME}
+            cd -
```

Comment:
Drop the `cd` commands, they are not needed if you use the correct path for the new link:

`ln -sf ${BOOT_CONFIG_MACHINE_EXTRA}-${IMAGE_IMXBOOT_TARGET} ${DEPLOYDIR}/${BOOT_NAME}`

Reply:
it's create a relative symlink, not absolute. Useful during artifactory on when build was done by docker with different path

Reply:
The link itself is unchanged, it is still relative. Only the location of the link is made absolute, which removes the need for the `cd` pair.

Reply:
@thochstein ah, sorry, I didn't notice that trick with force creation of symlink. Good to know 😃

Reply:
done